samples/vertex-vector-search/tests/system/workflow-test.py

```diff
@@ -0,0 +1,794 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import random
+import struct
+import threading
+import time
+from typing import Iterator, Callable
+
+import pytest
+from google.cloud import aiplatform_v1beta1, bigtable, workflows_v1
+from google.cloud.bigtable import column_family
+from google.cloud.workflows import executions_v1
+from google.cloud.workflows.executions_v1 import Execution
+
+# Configure Variables
+BIGTABLE_TABLE_NAME = "test_bigtable_vertex_integration_" + str(
+    random.randint(10000, 99999)
+)
+WORKFLOW_NAME = "test-bigtable-vvi-" + str(random.randint(10000, 99999))
+WORKFLOW_LOCATION = "us-central1"
+VERTEX_VECTOR_SEARCH_INDEX_ENDPOINT = (
+    "1597640674.us-central1-818418350420.vdb.vertexai.goog"
+)
+VERTEX_VECTOR_SEARCH_INDEX = "5500764314786594816"
+
+# Get the directory where this test file is located
+THIS_FILE_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+WORKFLOW_INPUT_FILE_PATH = THIS_FILE_DIRECTORY + "/workflow-input.json"
+WORKFLOW_YAML_FILE_PATH = (
+    os.path.dirname(os.path.dirname(THIS_FILE_DIRECTORY))
+    + "/workflows/batch-export.yaml"
+)
+
+
+# Define a custom log format
+log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+log_datefmt = "%Y-%m-%d %H:%M:%S"
+
+# Create a logger instance
+logger = logging.getLogger(__name__)
+
+# Set the log level to capture all log messages
+logger.setLevel(level=logging.DEBUG)
+
+# Create a handler for console output
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(logging.Formatter(log_format, log_datefmt))
+
+# Add the console handler and file handler to the logger
+logger.addHandler(console_handler)
+
+
+@pytest.fixture(scope="module")
+def project_id() -> Iterator[str]:
+    yield get_env_var("PROJECT_ID", "ID of the Cloud Bigtable project")
+
+
+@pytest.fixture(scope="module")
+def instance_id() -> Iterator[str]:
+    yield get_env_var("INSTANCE_ID", "ID of the Cloud Bigtable instance")
+
+
+@pytest.fixture(scope="module")
+def setup_workflow(project_id: str) -> Iterator[None]:
+    # Deploy Workflow
+    deploy_workflow(project_id, WORKFLOW_LOCATION, WORKFLOW_NAME)
+
+    yield
+
+    workflow_client = workflows_v1.WorkflowsClient()
+
+    workflow_full_path = (
+        "projects/{project}/locations/{location}/workflows/{workflow_name}".format(
+            project=project_id, location=WORKFLOW_LOCATION, workflow_name=WORKFLOW_NAME
+        )
+    )
+
+    # Initialize request argument(s)
+    request = workflows_v1.DeleteWorkflowRequest(
+        name=workflow_full_path,
+    )
+
+    # Make the request
+    operation = workflow_client.delete_workflow(request=request)
```

Comment:
Can you put this in a try/finally block? I want to make sure it is being cleaned up properly, even if it exits early

Reply:
Done